File: Task1/task1.go
package main
import (
	"fmt"
	"os"
	"strconv"
	"strings"
)
func main() {
	myString := "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
	var number int
	var input string
	fmt.Print("Введите число: ")
	fmt.Scanln(&input)
	for {
		var err error
		number, err = strconv.Atoi(input)
		if err == nil {
			break 
		}
		fmt.Println("Это не число. Попробуйте снова: ")
		fmt.Scanln(&input)
	}
	for number <= 0 || number > 26 {
		fmt.Println("Неправильное число. Попробуйте снова.")
		fmt.Scanln(&input)
		var err error
		number, err = strconv.Atoi(input)
		if err != nil {
			fmt.Println("Это не число. Пожалуйста, введите число: ")
			fmt.Scanln(&input)
			for {
				var err error
				number, err = strconv.Atoi(input)
				if err == nil {
					break
				}
				fmt.Println("Это не число. Попробуйте снова: ")
				fmt.Scanln(&input)
			}
		}
	}
	point := number - 1
	fmt.Println("Результат:")
	for point >= 0 {
		row := []string{}
		for i := number - 1; i >= point; i-- {
			row = append(row, string(myString[i]))
		}
		fmt.Println(strings.Join(row, " "))
		point--
	}
	os.Exit(0)
}

File: Task1/task1.rs
use std::io;
fn main() {
    let mystring = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    loop {
        let mut numbers = String::new();
        io::stdin().read_line(&mut numbers).unwrap();
        let number: i32 = match numbers.trim().parse() {
            Ok(num) => num,
            Err(_) => {
                println!("Неправильный ввод. Попробуйте снова: ");
                continue; 
            }
        };
        if number <= 0 || number > 26 {
            println!("Неправильное число. Попробуйте снова: ");
            continue; 
        }
        let mut point = number - 1;
        println!("Результат: ");
        while point >= 0 {
            for i in (point..number).rev() {
                print!("{}", mystring.chars().nth(i as usize).unwrap());
                print!(" ");
            }
            println!();
            point -= 1;
        }
        break; 
    }
}
